src/classes/PMT_PaymentCreator_TEST.cls

```diff
@@ -131,5 +133,44 @@ private with sharing class PMT_PaymentCreator_TEST {
         system.assertEquals(50, opmtWrittenOff[0].npe01__payment_amount__c);       
     }
 
+    /*********************************************************************************************************
+    * @description
+    operation:
+        close an Opp as Lost that has an existing payment and a null amount
+    verify:
+        payment is deleted, no infinite recursion takes place
+    **********************************************************************************************************/            
+    static testMethod void nullAmount() {
+        if (strTestOnly != '*' && strTestOnly != 'nullAmount') return;
+        
+        npe01__Contacts_And_Orgs_Settings__c PaymentsSettingsForTests = UTIL_CustomSettingsFacade.getContactsSettingsForTests(
+            new npe01__Contacts_And_Orgs_Settings__c (npe01__Payments_Enabled__c= true));
+        
+        // create test data
+        Account acct = new Account (
+            Name='Test Account'
+        );
+        insert acct;
+
+        Opportunity opp = new Opportunity (
+            Name='Test Opp',
+            AccountId = acct.id,
+            CloseDate = date.today(), 
+            Amount = 1,
+            StageName = UTIL_UnitTestData_TEST.getOpenStage()
+        );
+        insert opp;
+
+        list<npe01__OppPayment__c> opmtAll = [select id, npe01__payment_method__c, npe01__payment_amount__c, npe01__paid__c, npe01__written_off__c from npe01__OppPayment__c WHERE npe01__opportunity__c = :opp.Id];
+        system.assertEquals(1, opmtAll.size());
+
+        opp.Amount = null;
+        opp.StageName = 'Closed Lost';
```

Comment:
We have a "getClosedLostStage()" method in UTIL_UnitTestData_TEST that's commented out. I would have used it here if we had it; do you think I should add it back?